From 4067fd7814b81da5f2345a3b26f8223b515db4f4 Mon Sep 17 00:00:00 2001
From: betacentauri <@>
Date: Fri, 21 Jun 2013 12:31:42 +0200
Subject: [PATCH] qtdemux: don't assert if upstream size is not available when
 guessing bitrates

This patch is backported for version 0.10. Original version see here:
http://cgit.freedesktop.org/gstreamer/gst-plugins-good/commit/?id=e9682b938a56e52c168143d481f69028e5cf1f74

Fixes abort in push mode where the source is not seekable and the
size of the file is not available, as with

 cat foo.mp4 | gst-launch-1.0 playbin uri=fd://0

Less noticable with releases, since we disable all
g_assert() there.

https://bugzilla.gnome.org/show_bug.cgi?id=686008
---
 gst/isomp4/qtdemux.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index ea94cb2..45c859b 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -7591,7 +7591,7 @@ gst_qtdemux_guess_bitrate (GstQTDemux * qtdemux)
   GST_DEBUG_OBJECT (qtdemux, "Looking for streams with unknown bitrate");
 
   if (!gst_pad_query_peer_duration (qtdemux->sinkpad, &format, &size) ||
-      format != GST_FORMAT_BYTES) {
+      format != GST_FORMAT_BYTES || size <= 0) {
     GST_DEBUG_OBJECT (qtdemux,
         "Size in bytes of the stream not known - bailing");
     return;
@@ -7600,7 +7600,10 @@ gst_qtdemux_guess_bitrate (GstQTDemux * qtdemux)
   /* Subtract the header size */
   GST_DEBUG_OBJECT (qtdemux, "Total size %" G_GINT64_FORMAT ", header size %u",
       size, qtdemux->header_size);
-  g_assert (size >= qtdemux->header_size);
+
+  if (size < qtdemux->header_size)
+    return;
+
   size = size - qtdemux->header_size;
 
   if (!gst_qtdemux_get_duration (qtdemux, &duration) ||
-- 
1.7.9.5

